hack/upgrade-test.sh

```diff
@@ -0,0 +1,181 @@
+#
+# # Upgrade from the given version to the latest version in quay.io ( or the latest build in openshift-ci)
+# REGISTRY_NAMESPACE=sacharya IMAGE_TAG=0.0.1 IMAGE_REGISTRY=quay.io make upgrade-test
+#
+
+set -e
+
+source hack/common.sh
+
+IMAGE_REGISTRY="${IMAGE_REGISTRY:-quay.io}"
+REGISTRY_NAMESPACE="${REGISTRY_NAMESPACE:-}"
+IMAGE_NAME="ocs-registry"
+IMAGE_TAG="${IMAGE_TAG:-latest}"
+FULL_IMAGE_NAME="${IMAGE_REGISTRY}/${REGISTRY_NAMESPACE}/${IMAGE_NAME}:${IMAGE_TAG}"
+SUBSCRIBE_CHANNEL="${SUBSCRIBE_CHANNEL:alpha}"
+DEPLOY_MANIFEST=$OUTDIR_CLUSTER_DEPLOY_MANIFESTS/deploy-with-olm.yaml
+
+mkdir -p $OUTDIR_CLUSTER_DEPLOY_MANIFESTS
+
+cp deploy/deploy-with-olm.yaml $DEPLOY_MANIFEST
+
+sed -i "s|image\: .*/ocs-registry:.*$|image: ${FULL_IMAGE_NAME}|g" $DEPLOY_MANIFEST
+
+echo "Applying ocs-operator manifests"
+oc apply -f $DEPLOY_MANIFEST
+
+# Wait for ocs-operator, rook operator and noobaa operator to come online
+deployments=(ocs-operator rook-ceph-operator noobaa-operator)
+
+for i in ${deployments[@]}; do
+	current_time=0
+	sample=10
+	timeout=1600
+	while [ -z "$(oc get deployments -n openshift-storage | grep "${i} ")" ]; do
+		echo "Waiting for deployment ${i} to be created..."
+		sleep $sample
+		current_time=$((current_time + sample))
+		if [[ $current_time -gt $timeout ]]; then
+			# On failure, dump debug info about the ocs-catalogsource
+			echo "----------- dumping all pods for debug ----------------"
+			oc get pods --all-namespaces
+			echo "----------- dumping all pods for debug ----------------"
+			echo ""
+			echo "ERROR: Timed out waiting for deployment ${i} to be created"
+			exit 1
+		fi
+	done
+	echo "Deployment ${i} has been created"
+done
+
+for i in ${deployments[@]}; do
+	echo "Waiting for deployment $i to become available"
+	oc wait -n openshift-storage deployment ${i} --for condition=Available --timeout 12m
+done
+
+csv_name=$(oc get csvs -n openshift-storage | grep ocs-operator | head -1 | awk '{print $1}')
+echo "Installed CSV $csv_name"
+
+echo "Success! Deployed $FULL_IMAGE_NAME"
+echo "Starting the upgrade process now"
+
+# This image should point to the latest version in quay.io
+# Or the latest build in openshift-ci environment.
+UPGRADE_VERSION_IMAGE_REGISTRY="${UPGRADE_VERSION_IMAGE_REGISTRY:-quay.io}"
+UPGRADE_VERSION_REGISTRY_NAMESPACE="${UPGRADE_VERSION_REGISTRY_NAMESPACE:-}"
+UPGRADE_VERSION_IMAGE_NAME="ocs-registry"
+UPGRADE_VERSION_IMAGE_TAG="${UPGRADE_VERSION_IMAGE_TAG:-latest}"
+UPGRADE_VERSION_FULL_IMAGE_NAME="${UPGRADE_VERSION_IMAGE_REGISTRY}/${UPGRADE_VERSION_REGISTRY_NAMESPACE}/${UPGRADE_VERSION_IMAGE_NAME}:${UPGRADE_VERSION_IMAGE_TAG}"
+UPGRADE_VERSION_SUBSCRIBE_CHANNEL="${UPGRADE_VERSION_SUBSCRIBE_CHANNEL:alpha}"
+
+# Override the image name when this is invoked from openshift ci
+if [ -n "$OPENSHIFT_BUILD_NAMESPACE" ]; then
+	UPGRADE_VERSION_FULL_IMAGE_NAME="registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:ocs-registry"
```

Comment:
can we parameterize usage of "registry.svc.ci.openshift.org"? I thought that's why the IMAGE_REGISTRY envvar exists.